src/components/ui/BaseInput.vue

```diff
@@ -1,17 +1,33 @@
 <template>
-	<div>
+	<div class="root">
 		<label class="input__wrapper">
 			<span :class="{ label: true, active: value && value !== 0 }">
 				{{ label }}
 			</span>
+			<FlatPickr
+				v-if="type === 'date'"
+				v-model="value"
+				v-bind="$attrs"
+				:config="config"
+				:wrap="true"
+				v-on="$listeners"
+				@input="$emit('update', value)"
+			>
+				<input
+					:type="type"
```

Comment:
type="date" can be hardcoded.
```suggestion
					type="date"
```